File: explorer/src/main/java/com/duongame/archive/Zip4jFile.java
package com.duongame.archive;

import net.lingala.zip4j.core.ZipFile;
import net.lingala.zip4j.exception.ZipException;
import net.lingala.zip4j.model.FileHeader;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by namjungsoo on 2018-01-28.
 */

public class Zip4jFile implements IArchiveFile {
    ZipFile file;

    public Zip4jFile(String zipPath) {
        try {
            file = new ZipFile(zipPath);
        } catch (ZipException e) {
            e.printStackTrace();
            file = null;
        }
    }

    @Override
    public boolean isEncryped() {
        if (file != null) {
            try {
                return file.isEncrypted();
            } catch (Exception e) {

            }
        }
        return false;
    }

    @Override
    public void setPassword(String password) {
        if(file != null) {
            try {
                file.setPassword(password);
            } catch (ZipException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    public ArrayList<ArchiveHeader> getHeaders() {
        if (file != null) {
            try {
                List<FileHeader> headers = file.getFileHeaders();
                if (headers == null)
                    return null;

                ArrayList<ArchiveHeader> newHeaders = new ArrayList<>();
                for (FileHeader header : headers) {
                    newHeaders.add(new ArchiveHeader(header.getFileName(), header.getUncompressedSize()));
                }
                return newHeaders;
            } catch (ZipException e) {
                e.printStackTrace();
                return null;
            }
        }
        return null;
    }

    @Override
    public boolean extractFile(String fileName, String destPath) {
        try {
            file.extractFile(fileName, destPath);
            return true;
        } catch (ZipException e) {
            e.printStackTrace();
        }
        return false;
    }

    @Override
    public boolean extractAll(String destPath) {
        try {
            file.extractAll(destPath);
            return true;
        } catch (ZipException e) {
            e.printStackTrace();
        }
        return false;
    }

    @Override
    public void destroy() {

    }
}

File: Explorer/src/main/java/com/duongame/archive/Zip4jFile.kt
package com.duongame.archive

import net.lingala.zip4j.core.ZipFile
import net.lingala.zip4j.exception.ZipException
import net.lingala.zip4j.model.FileHeader
import java.util.*

/**
 * Created by namjungsoo on 2018-01-28.
 */
class Zip4jFile(zipPath: String) : ArchiveFile {
    var file: ZipFile = ZipFile(zipPath)

    override val isEncryped: Boolean = file.isEncrypted

    override fun setPassword(password: String) {
        file.setPassword(password)
    }

    override fun setFileNameCharset(charset: String) {
        file.setFileNameCharset(charset)
    }

    override val headers: ArrayList<ArchiveHeader>?
        get() {
            try {
            val headers = file.fileHeaders as List<FileHeader>? ?: return null
            val newHeaders = ArrayList<ArchiveHeader>()
            for (header in headers) {
                newHeaders.add(ArchiveHeader(header.fileName, header.uncompressedSize))
            }
            return newHeaders
        } catch (e: ZipException) {
            e.printStackTrace()
            return null
        }
    }

    override fun extractFile(fileName: String, destPath: String): Boolean {
        try {
            file.extractFile(fileName, destPath)
            return true
        } catch (e: ZipException) {
            e.printStackTrace()
        }
        return false
    }

    override fun extractAll(destPath: String): Boolean {
        try {
            file.extractAll(destPath)
            return true
        } catch (e: ZipException) {
            e.printStackTrace()
        }
        return false
    }

    override fun destroy() {}
}